Cython/Utility/ModuleSetupCode.c

```diff
@@ -1485,7 +1485,7 @@ static CYTHON_INLINE float __PYX_NAN() {
 
 /////////////// UtilityFunctionPredeclarations.proto ///////////////
 
-typedef struct {PyObject **p; const char *s; const Py_ssize_t n; const char* encoding;
+typedef struct {const char *s; const Py_ssize_t n; const char* encoding;
                 const char is_unicode; const char is_str; const char intern; } __Pyx_StringTabEntry; /*proto*/
```

Comment:
We can probably reduce the size of this struct by using `int` instead of `Py_ssize_t` for the length if possible. We could template-generate this and insert the longest length of all the strings, and then doing
```c
typedef struct {const char *s; const char* encoding;
#if INT_MAX >= %(maxlength)d
    const int n;
#elif LONG_MAX >= %(maxlength)d
    const long n;
#else
    const Py_ssize_t n;
#endif
    const char is_unicode; const char is_str; const char intern; } __Pyx_StringTabEntry; /*proto*/
```
I've also reordered the fields (pointers first) to avoid padding issues.

Reply:
This doesn't feel worth the effort to me.  If you want to do it yourself as part of this change then go ahead though (or as a follow up)

Reply:
I'll try it as a follow-up. I've seen really string heavy modules in the past. We can probably also use a name map for the encodings (there'll usually be only one – or zero), and store it as another `char` index, thus going from an initial 32 bytes down to 16 per string.

Reply:
FWIW, you may also want to pack the structure if you're concerned about binary size, and also use bitfields for the various flags.
https://stackoverflow.com/a/3312896/10194

Reply:
See https://github.com/cython/cython/commit/368a750952f97042f3e8728f18c350f80413c84d